Add large grants to My grants and add large grant detail page

damianmarti opened the pull request:
- UCDC grants are shown on My Grants page

![localhost_3000_my-grants (1)](https://github.com/user-attachments/assets/f763753d-7421-433d-abd6-046000a85f85)

- Created USDC detail page for user

![localhost_3000_large-grants_95 (6)](https://github.com/user-attachments/assets/ba0e40a6-b866-4f0f-acdf-af0f917bd8a1)

I used the same progress bar than the one used on the home page. Only counting funds received. We can improve it later if we want to show how many milestones were completed.

- Mobile view

![localhost_3000_large-grants_95 (8)](https://github.com/user-attachments/assets/956e58a4-9cf2-43ca-bc58-73e1fc91965d)

- Complete milestone modal (feature not implemented yet, issue created #35 )

![localhost_3000_large-grants_95](https://github.com/user-attachments/assets/e4c48369-0d1a-41fa-a4bc-05ea26d84533)

- Improvements to the schema and the seed data.

closes #24

rin-st commented:
I think we need add year here, and same for "Paid on"
<img width="213" alt="image" src="https://github.com/user-attachments/assets/9b12b835-75c8-4a8f-bf5e-b8b8966b2756" />

Also I think checkmark icon should be different here, probably yellow, or with info icon with tooltip 
<img width="665" alt="Screenshot 2025-03-19 at 15 35 46" src="https://github.com/user-attachments/assets/88f36e10-2903-4aa4-8678-827ebfa4cd4f" />

And question, do we need to make amounts like 2,222 ? You can use [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#using_tolocalestring)
<img width="157" alt="image" src="https://github.com/user-attachments/assets/6f67fc53-6aab-476d-bda1-d10a9a9d1069" />

rin-st reviewed (commented):
Great job Damu! Added some comments

damianmarti commented on packages/nextjs/app/api/builders/[builderAddress]/grants/route.ts:
```diff
@@ -19,8 +21,13 @@ export async function GET(_request: Request, { params }: { params: { builderAddr
 
-    const grants = await getBuilderGrants(builderAddress);
-    return NextResponse.json(grants, { status: 200 });
+    const grants = (await getBuilderGrants(builderAddress)).map(grant => ({ ...grant, type: "grant" }));
```
I think we use almost all this information. And since this endpoint is only available to the builder, we are not exposing data to others.

damianmarti commented on packages/nextjs/components/pg-ens/BadgeMilestone.tsx:
```diff
@@ -0,0 +1,50 @@
+  proposed: "bg-warning",
+  rejected: "bg-error",
+  completed: "bg-warning",
+  paid: "bg-success",
```
I changed some stuff, like colors and text to others' statuses too, so I think it's simpler if we keep two separate components.

damianmarti commented:
> I think we need add year here, and same for "Paid on" <img alt="image" width="213" src="https://private-user-images.githubusercontent.com/25638585/424475956-9b12b835-75c8-4a8f-bf5e-b8b8966b2756.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzU5NTYtOWIxMmI4MzUtNzVjOC00YThmLWJmNWUtYjhiODk2NmIyNzU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmMmU0OWVlNTZhYzBmNWZjNGQ0YzJiNzQ3MjUzYmUxNGU0YWNlMTQ3NGI5ZmVhNDY5Zjc0OTFlNDdmNGZhNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._NF23NHCdH3VYNXFPesjx0nkswD70XT67sofiQEhgb0">
> 

Yes, makes sense. I'm using the same getFormattedDate function used for ETH grants. Do you think we should add the year there?

> Also I think checkmark icon should be different here, probably yellow, or with info icon with tooltip <img alt="Screenshot 2025-03-19 at 15 35 46" width="665" src="https://private-user-images.githubusercontent.com/25638585/424476409-88f36e10-2903-4aa4-8678-827ebfa4cd4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzY0MDktODhmMzZlMTAtMjkwMy00YWE0LTg2NzgtODI3ZWJmYTRjZDRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZWJiZGJmY2RmNWVkZGE5ZTc4NjNkNDUzZTVkNmQ0MDJjZjA0ODI3NDc3YmEwMDlmNDI3YzRjYTU4Y2ZkYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y20yKAsmayA30veM0mX93TzIGqIVW8tIq22s41yROPM">

Yes, thanks, I will change it, and maybe we can improve the style later.

> 
> And question, do we need to make amounts like 2,222 ? You can use [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#using_tolocalestring) <img alt="image" width="157" src="https://private-user-images.githubusercontent.com/25638585/424478127-6f67fc53-6aab-476d-bda1-d10a9a9d1069.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzgxMjctNmY2N2ZjNTMtNmFhYi00NzZkLWJkYTEtZDEwYTlhOWQxMDY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MGY1MjkxM2M4NTE3NGUwZmZiZTcwY2ZiYTI3ODYxYTNmZWYzNTkyZWI1NTczZTNmOTc2YTVjY2M4YTcxZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NWdsYZ6DfjqtX5FqSeUmme78LrkHcjDBnm2SXUbAYu4">

Yes! I'm using Intl.NumberFormat().format in the apply page but using toLocaleString should works too.

damianmarti commented:
> > I think we need add year here, and same for "Paid on" <img alt="image" width="213" src="https://private-user-images.githubusercontent.com/25638585/424475956-9b12b835-75c8-4a8f-bf5e-b8b8966b2756.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzU5NTYtOWIxMmI4MzUtNzVjOC00YThmLWJmNWUtYjhiODk2NmIyNzU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmMmU0OWVlNTZhYzBmNWZjNGQ0YzJiNzQ3MjUzYmUxNGU0YWNlMTQ3NGI5ZmVhNDY5Zjc0OTFlNDdmNGZhNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._NF23NHCdH3VYNXFPesjx0nkswD70XT67sofiQEhgb0">
> 
> Yes, makes sense. I'm using the same getFormattedDate function used for ETH grants. Do you think we should add the year there?

Oh, I forgot this is the Month and year only. We can use another date format function here, maybe just toLocaleString

> 
> > Also I think checkmark icon should be different here, probably yellow, or with info icon with tooltip <img alt="Screenshot 2025-03-19 at 15 35 46" width="665" src="https://private-user-images.githubusercontent.com/25638585/424476409-88f36e10-2903-4aa4-8678-827ebfa4cd4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzY0MDktODhmMzZlMTAtMjkwMy00YWE0LTg2NzgtODI3ZWJmYTRjZDRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZWJiZGJmY2RmNWVkZGE5ZTc4NjNkNDUzZTVkNmQ0MDJjZjA0ODI3NDc3YmEwMDlmNDI3YzRjYTU4Y2ZkYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y20yKAsmayA30veM0mX93TzIGqIVW8tIq22s41yROPM">
> 
> Yes, thanks, I will change it, and maybe we can improve the style later.
> 
> > And question, do we need to make amounts like 2,222 ? You can use [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#using_tolocalestring) <img alt="image" width="157" src="https://private-user-images.githubusercontent.com/25638585/424478127-6f67fc53-6aab-476d-bda1-d10a9a9d1069.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzgxMjctNmY2N2ZjNTMtNmFhYi00NzZkLWJkYTEtZDEwYTlhOWQxMDY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MGY1MjkxM2M4NTE3NGUwZmZiZTcwY2ZiYTI3ODYxYTNmZWYzNTkyZWI1NTczZTNmOTc2YTVjY2M4YTcxZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NWdsYZ6DfjqtX5FqSeUmme78LrkHcjDBnm2SXUbAYu4">
> 
> Yes! I'm using Intl.NumberFormat().format in the apply page but using toLocaleString should works too.

damianmarti commented on packages/nextjs/app/api/builders/[builderAddress]/grants/route.ts:
```diff
@@ -1,9 +1,11 @@
 import { authOptions } from "~~/utils/auth";
 
-export type BuilderGrantsResponse = Awaited<ReturnType<typeof getBuilderGrants>>;
+export type BuilderGrantsResponse = DiscriminatedBuilderGrant[]; //Awaited<ReturnType<typeof getBuilderGrants>>;
```
Removed!

damianmarti commented on packages/nextjs/app/large-grants/[grantId]/_components/ProjectTimeline/TimelineFakeStage.tsx:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <div className="grid sm:grid-cols-2">
+      <div
+        className={`hidden sm:block border-primary border-secondary border-dashed ${
```
Fixed!

damianmarti commented on packages/nextjs/app/large-grants/[grantId]/_components/ProjectTimeline/index.tsx:
```diff
@@ -0,0 +1,32 @@
+        {grant.stages.map(stage => (
+          <TimelineStage key={stage.stageNumber} stage={stage} />
+        ))}
+        {(() => {
```
Changed! Please check it.

damianmarti commented:
> > > I think we need add year here, and same for "Paid on" <img alt="image" width="213" src="https://private-user-images.githubusercontent.com/25638585/424475956-9b12b835-75c8-4a8f-bf5e-b8b8966b2756.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzU5NTYtOWIxMmI4MzUtNzVjOC00YThmLWJmNWUtYjhiODk2NmIyNzU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmMmU0OWVlNTZhYzBmNWZjNGQ0YzJiNzQ3MjUzYmUxNGU0YWNlMTQ3NGI5ZmVhNDY5Zjc0OTFlNDdmNGZhNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._NF23NHCdH3VYNXFPesjx0nkswD70XT67sofiQEhgb0">
> > 
> > 
> > Yes, makes sense. I'm using the same getFormattedDate function used for ETH grants. Do you think we should add the year there?
> 
> Oh, I forgot this is the Month and year only. We can use another date format function here, maybe just toLocaleString

Used toLocaleDateString for milestone deadline and toLocaleString for paid field.


> 
> > > Also I think checkmark icon should be different here, probably yellow, or with info icon with tooltip <img alt="Screenshot 2025-03-19 at 15 35 46" width="665" src="https://private-user-images.githubusercontent.com/25638585/424476409-88f36e10-2903-4aa4-8678-827ebfa4cd4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzY0MDktODhmMzZlMTAtMjkwMy00YWE0LTg2NzgtODI3ZWJmYTRjZDRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZWJiZGJmY2RmNWVkZGE5ZTc4NjNkNDUzZTVkNmQ0MDJjZjA0ODI3NDc3YmEwMDlmNDI3YzRjYTU4Y2ZkYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y20yKAsmayA30veM0mX93TzIGqIVW8tIq22s41yROPM">
> > 
> > 
> > Yes, thanks, I will change it, and maybe we can improve the style later.

Changed!


> > > And question, do we need to make amounts like 2,222 ? You can use [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#using_tolocalestring) <img alt="image" width="157" src="https://private-user-images.githubusercontent.com/25638585/424478127-6f67fc53-6aab-476d-bda1-d10a9a9d1069.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTUyMzEsIm5iZiI6MTc0MjM5NDkzMSwicGF0aCI6Ii8yNTYzODU4NS80MjQ0NzgxMjctNmY2N2ZjNTMtNmFhYi00NzZkLWJkYTEtZDEwYTlhOWQxMDY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDE0MzUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MGY1MjkxM2M4NTE3NGUwZmZiZTcwY2ZiYTI3ODYxYTNmZWYzNTkyZWI1NTczZTNmOTc2YTVjY2M4YTcxZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NWdsYZ6DfjqtX5FqSeUmme78LrkHcjDBnm2SXUbAYu4">
> > 
> > 
> > Yes! I'm using Intl.NumberFormat().format in the apply page but using toLocaleString should works too.

Changed!

damianmarti commented:
> Great job Damu! Added some comments

Thanks @rin-st for the review! All the comments should be fixed now.

rin-st commented:
> Yes, makes sense. I'm using the same getFormattedDate function used for ETH grants. Do you think we should add the year there?

Oh sorry, I though it 25th februray, not Feb 2025. 

> Used toLocaleDateString for milestone deadline and toLocaleString for paid field.
> Changed!

Could you add screenshot of how it looks now (Or steps to reproduce/seed data)? so I could check it

damianmarti commented:
> > Yes, makes sense. I'm using the same getFormattedDate function used for ETH grants. Do you think we should add the year there?
> 
> Oh sorry, I though it 25th februray, not Feb 2025.
> 
> > Used toLocaleDateString for milestone deadline and toLocaleString for paid field.
> > Changed!
> 
> Could you add screenshot of how it looks now (Or steps to reproduce/seed data)? so I could check it

yes, sure!

![localhost_3000_large-grants_104](https://github.com/user-attachments/assets/065de44a-2573-43b7-9413-1e538a118c98)

You can change the builder address from the seed data to your burner wallet address and you can see all the seed grants on My Grants.
